production/helm/promtail/templates/daemonset.yaml

```
@@ -44,11 +44,13 @@ spec:
           imagePullPolicy: {{ .Values.image.pullPolicy }}
           args:
             - "-config.file=/etc/promtail/promtail.yaml"
+            {{- if not .Values.config.clients }}
```

I would suggest this because you can use either `client`(deprecated) or `clients` in the config. If someone is using `client` with `url` set then they would face same problem as you. In case of `clients`, since we can't verify whether all the configs have `url` set or not, we can just assume `url` would have been set for each of them since user is willing to send logs to multiple different targets.

```suggestion
            {{- if not (or .Values.config.client.url .Values.config.clients) }}
```